backend/match-reporter/bot-factory.js

```diff
@@ -17,24 +17,27 @@ class SingleChannelBot {
 }
 
 const makeBot = (botToken, channelName) => {
-  if (!botToken || !channelName) {
-    throw Error('botToken or channelName missing')
-  }
-  const slackbot = new SlackBot({
-    token: botToken
-  })
-
   return new Promise((resolve, reject) => {
+    if (!botToken || !channelName) {
+      reject('botToken or channelName missing')
+    }
+
+    const slackbot = new SlackBot({
+      token: botToken
+    })
+
+    slackbot.on('error', (error) => reject(error.message))
+
     slackbot.on('start', async () => {
       let groupId 
       try {
         groupId = await slackbot.getGroupId(channelName)
       } catch (error) {
-        return reject(error)
+        reject(error.message)
```

Comment:
1. why are you rejecting with a string? i thought it's better to reject with a stack trace. more so since you're catching it later on (throwing non Error types is an antipattern)
2. why did you remove the return from revolse and reject? i couldn't find it after a quick search, but is it in the spec, what happens  if you call one of the functions multiple times?

Reply:
1. Well, when you reject, you don't really throw... But sure, we can reject with `Error`s.
2. My bad. I'll fix it. Though the `return` doesn't make much sense there IMO, so I guess I'll fix it by resolving within `try` rather than returning `return` back. ;)

Reply:
1.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject#Description
whether
you throw it or not is irrelevant, according to this, rejected reason
should be an error

Reply:
That's not what the documentation says. It says it may be useful for it to be an instance of `Error`, not that it has to be that. Anyway, I changed it.

Reply:
```code
const makeBot = async (botToken, channelName) => {
	if(!botToken || !channelName) throw new Error('botToken or channelName missing');
	const slackbot = new SlackBot({ token: botToken });
	return Promise.race([
		new Promise((resolve, reject) => slackbot.on('error', error => reject(error)),
		new Promise((resolve, reject) => slackbot.on('start', () =>
                   slackbot.getGroupId(channelName)
                   .then(groupId => resolve(new SingleChannelBot(slackbot, channelName, groupId)))
                   .catch(error => reject(error))
		)
	]);
}
```
`new Promise()` only the callbacks dammit :dagger: :dagger: :D

Reply:
@pilartomas This would mean that you'd have to catch errors in the promise chain as well as surround calling `makeBot` with a `try/catch` block to catch the error you throw on the first line. I think it's better to handle the errors only in one place. Anyway, I like that `race` thing. Feel free to create a PR with that change. ;) I am done here. :D

Reply:
@zbrcz Nope for the first sentence. Throwing from `async` function (I made `makeBot` async) gets compiled into rejection. The code should work same as the old one from the caller's viewpoint :)

Reply:
Ah, I didn't notice that you made it async... Anyway, as I said, feel free to create a PR. ;)